### ISSUE TITLE
[studio-ui] Add date format support to Crafter Studio 

### ISSUE THREAD

**Issue by russdanner** [craftercms/craftercms#3757]
**Is your feature request related to a problem? Please describe.**
Different countries use different date formats. Authors located in those countries find adapting to Studio's US-based date formats counter-intuitive. Example: Authors in the UK and UK influenced regions of the world are accustomed to a UK date format (day-month-year) and would like the ability to opt-in to that format within the Studio UI for dashboards and date pickers. Date and time notation in the United Kingdom records the date using the day-month-year format (21 October 2011 or 21/10/11).

_new_

Also, support the 24-hour time format.

**Describe the solution you'd like**
Enable a site-wide setting for date format to be used in API responses, dashboards and on date pickers in the Crafter UI.

**Additional context**
https://en.wikipedia.org/wiki/Date_format_by_country

**Comment by odlg** [comment]
Does this also include 24hour support? Or should there be a separate issue for that?

**Comment by sumerjabri** [comment]
@odlg I added the 24-hour time format to this ticket.

**Comment by odlg** [comment]
Thanks, 24 hour time is used in large parts of the world - see the map on this page: https://en.wikipedia.org/wiki/Date_and_time_representation_by_country

**Comment by rart** [comment]
Work done; waiting on https://github.com/craftercms/craftercms/issues/4635 to finalize the Global app piece. 
Testing can begin.

**Comment by russdanner** [comment]
I have not found a way to change the date format
<img width="1252" alt="Screen Shot 2021-11-12 at 9 56 06 AM" src="https://user-images.githubusercontent.com/169432/141487207-f546062a-4e2f-4d53-b317-923110091c36.png">
s?

**Comment by rart** [comment]
I can't find a way to answer your previous comment.
What are you trying to do? 
I don't know what `s?` means.

Change your `<localeCode />` if you're trying to view the date in as in a _different locale_.

**Comment by russdanner** [comment]
<img width="1377" alt="Screen Shot 2021-11-30 at 8 44 29 PM" src="https://user-images.githubusercontent.com/169432/144156587-2051cfed-afb5-47ef-862c-75f3634711a4.png">
<img width="1440" alt="Screen Shot 2021-11-30 at 8 43 38 PM" src="https://user-images.githubusercontent.com/169432/144156590-6557ed49-617b-4639-8a1f-dfb94f346c1b.png">

When I add a locale like BST is time adjust but not the format. How do I adjust the format so that dates are formatted in a European fashion (for example) This request is about adding date format support but I am not able to format the date.

**Comment by rart** [comment]
As I said above, change the `<localeCode />`.

**Comment by russdanner** [comment]
I sorry I target fixated on the timezone code and didn't notice the option above.

**Comment by russdanner** [comment]
Confirmed
